lib/ansiblelint/__main__.py

```diff
@@ -56,6 +59,20 @@ def initialize_logger(level: int = 0) -> None:
     _logger.debug("Logging initialized to level %s", logging_level)
 
 
+def choose_formatter_factory(
+    options_list: "Namespace"
+) -> Type[formatters.BaseFormatter]:
+    """Select an output formatter based on the incoming command line arguments."""
+    r: Type[formatters.BaseFormatter] = formatters.Formatter
+    if options_list.quiet:
+        r = formatters.QuietFormatter
+    elif options_list.parseable:
+        r = formatters.ParseableFormatter
+    elif options_list.parseable_severity:
+        r = formatters.ParseableSeverityFormatter
+    return r
```

Comment:
Extra var is unnecessary but instead using a TypeVar is needed as requested @ https://github.com/ansible/ansible-lint/pull/931/files#r463479262.

Reply:
I tried different approaches and they did not work. If you know a better way to do it, one that is passing all our testing, lets do it in a follow-up.

Reply:
I can take a look in a week.